src/test/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderBookingTest.kt

```diff
@@ -0,0 +1,628 @@
+        .bookNumber("R1234Q")
+        .build().also {
+          offender.addBooking(it)
+
+          it.addExternalMovement(
+            ExternalMovement.builder()
+              .movementType(MovementType("ADM", "Admission"))
+              .movementDirection(MovementDirection.IN)
+              .movementReason(MovementReason("B", "Recall"))
+              .movementTime(LocalDateTime.of(2021, 1, 4, 9, 30))
+              .toAgency(AgencyLocation.builder().id("MDI").build())
+              .build()
+          )
+          it.addExternalMovement(
+            ExternalMovement.builder()
+              .movementType(MovementType("CRT", "Court"))
+              .movementDirection(MovementDirection.OUT)
+              .movementReason(MovementReason("CRT", "Court Appearance"))
+              .movementTime(LocalDateTime.of(2021, 1, 15, 9, 30))
+              .build()
+          )
+          it.addExternalMovement(
+            ExternalMovement.builder()
+              .movementType(MovementType("CRT", "Court"))
+              .movementDirection(MovementDirection.IN)
+              .movementReason(MovementReason("CRT", "Court Appearance"))
+              .movementTime(LocalDateTime.of(2021, 1, 15, 15, 30))
+              .build()
+          )
+          it.addExternalMovement(
+            ExternalMovement.builder()
+              .movementType(MovementType("REL", "Release"))
+              .movementDirection(MovementDirection.OUT)
+              .movementReason(MovementReason("HP", "Hospital"))
+              .movementTime(LocalDateTime.of(2021, 2, 28, 15, 30))
+              .build()
+          )
+        }
+
+      with(offender.prisonerInPrisonSummary) {
+        assertThat(prisonPeriod).hasSize(3)
+        assertThat(prisonPeriod[0].bookNumber).isEqualTo("R1234K")
+        assertThat(prisonPeriod[1].bookNumber).isEqualTo("R1234T")
+        assertThat(prisonPeriod[2].bookNumber).isEqualTo("R1234Q")
+        assertThat(prisonPeriod[0].movementDates).hasSize(1)
+        assertThat(prisonPeriod[1].movementDates).hasSize(2)
+        assertThat(prisonPeriod[2].movementDates).hasSize(1)
+        assertThat(prisonPeriod[0].entryDate).isEqualTo(LocalDateTime.of(2019, 1, 4, 9, 30))
+        assertThat(prisonPeriod[0].releaseDate).isEqualTo(LocalDateTime.of(2019, 2, 28, 15, 30))
+        assertThat(prisonPeriod[0].prisons).containsExactly("WWI")
+        assertThat(prisonPeriod[1].entryDate).isEqualTo(LocalDateTime.of(2020, 1, 4, 9, 30))
+        assertThat(prisonPeriod[1].releaseDate).isEqualTo(LocalDateTime.of(2020, 2, 28, 15, 30))
+        assertThat(prisonPeriod[1].prisons).containsExactly("MDI")
+        assertThat(prisonPeriod[2].entryDate).isEqualTo(LocalDateTime.of(2021, 1, 4, 9, 30))
+        assertThat(prisonPeriod[2].releaseDate).isEqualTo(LocalDateTime.of(2021, 2, 28, 15, 30))
+        assertThat(prisonPeriod[2].prisons).containsExactly("MDI")
+      }
+    }
+
+    @DisplayName(value = "cope with bad data and no admission")
```

Comment:
this is a new test - currently copy of the test above but need to change in next PR